common/headtracker/head_tracker.go

```diff
@@ -346,9 +345,9 @@ func (ht *HeadTracker[HTH, S, ID, BLOCK_HASH]) backfill(ctx context.Context, hea
 	return
 }
 
-func (ht *HeadTracker[HTH, S, ID, BLOCK_HASH]) fetchAndSaveHead(ctx context.Context, n int64) (HTH, error) {
-	ht.log.Debugw("Fetching head", "blockHeight", n)
-	head, err := ht.client.HeadByNumber(ctx, big.NewInt(n))
+func (ht *HeadTracker[HTH, S, ID, BLOCK_HASH]) fetchAndSaveHead(ctx context.Context, n int64, hash BLOCK_HASH) (HTH, error) {
+	ht.log.Debugw("Fetching head", "blockHeight", n, "blockHash", hash)
+	head, err := ht.client.HeadByHash(ctx, hash)
```

Comment:
It makes sense that we are using HeadByHash instead to make sure that the head we are getting is the unique one. 

I am curious to see if other chains who does not have HeadByHash implementation like solana would implement this

Reply:
Interesting. Solana's case does complicate things, but for now, I would rather fix it for the ETH side, and think about Solana later. Are there other chains that don't support `HeadByHash` ?

Reply:
Cosmos supports HeadByHash, starknet i think as well.